DeliveryCompany.BusinessLayer/PackageService.cs

```diff
@@ -65,21 +76,54 @@ private Package CoordinateAssignment(Package package)
             return package;
         }
 
-        public void Update(Package package)
+        public async Task UpdateAsync(Package package)
         {
             using (var context = _deliveryCompanyDbContextFactoryMethod())
             {
                 context.Packages.Update(package);
-                context.SaveChanges();
+                await context.SaveChangesAsync();
             }
         }
 
+        public async Task UpdateByIdWithNotifyAsync(int id, Package package)
+        {
+            package.Id = id;
+
+            using (var context = _deliveryCompanyDbContextFactoryMethod())
+            {
+                context.Packages.Update(package);
+                await context.SaveChangesAsync();
+            }
+
+            _notificationService.NotifyOfPackageDelivery(package);
+        }
+
         public void UpdatePackages(List<Package> packages, StateOfPackage stateOfPackage)
```

Comment:
Just wonder - was there a requirement to not notify when package is delivered anywhere? Additionally, even if there was, method UpdateAsync and UpdateByIdWithNotifyAsync can be one, with optional parameter `public async Task UpdateAsync(Package package, bool notify = false){ ... }`